Firestore/core/src/util/task.h

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_UTIL_TASK_H_
+#define FIRESTORE_CORE_SRC_UTIL_TASK_H_
+
+#include <atomic>
+#include <condition_variable>  // NOLINT(build/c++11)
+#include <memory>
+#include <mutex>   // NOLINT(build/c++11)
+#include <thread>  // NOLINT(build/c++11)
+
+#include "Firestore/core/src/util/executor.h"
+
+namespace firebase {
+namespace firestore {
+namespace util {
+
+// For testing
+class TrackingTask;
```

Comment:
IIRC, a friend declaration serves as a forward declaration as well, so this can be omitted.